proxy.go

```diff
@@ -113,6 +113,16 @@ func (p *Proxy) SetMITM(config *mitm.Config) {
 	p.mitm = config
 }
 
+// SetSkipTLSVerify configures whether the proxy skips verification of server certificates.
+func (p *Proxy) SetSkipTLSVerify(skip bool) {
```

Comment:
I would just do this part in the `main.go` rather than in the proxy itself. Just make a new `http.Transport` with `TLSClientConfig`, set `InsecureSkipVerify` to whatever the flag is, and set the RoundTripper on the proxy.